### PR TITLE
update(release): replace `sed` command with node functions

### PR DESCRIPTION
**Pull request by robertmesserle**
I haven't tested this yet, so probably not ready to merge - but these are the commands pulled from `snapshot-docs-site.sh` and escaped for JS.

### CODE REVIEW

**Comment by robertmesserle** [comment]
@topherfangio

File: release.js [hunk 1]
```diff
@@ -266,9 +266,9 @@
     exec([
            'rm -rf dist',
            'gulp docs',
-           'sed -i \'\' \'s,http:\\/\\/localhost:8080\\/angular-material,https:\\/\\/cdn.gitcdn.xyz/cdn/angular/bower-material/v{{newVersion}}/angular-material,g\' dist/docs/docs.js',
-           'sed -i \'\' \'s,http:\\/\\/localhost:8080\\/docs\\.css,https:\\/\\/material.angularjs.org/{{newVersion}}/docs.css,g\' dist/docs/docs.js',
-           'sed -i \'\' \'s,base\ href=\\",base\ href=\\"/{{newVersion}},g\' dist/docs/index.html'
+           'sed -i "s,http://localhost:8080/angular-material,https://cdn.gitcdn.xyz/cdn/angular/bower-material/v{{newVersion}}/angular-material,g" dist/docs/docs.js',
+           'sed -i "s,http://localhost:8080/docs.css,https://material.angularjs.org/$VERSION/docs.css,g" dist/docs/docs.js',
```
Comment (topherfangio) [inline release.js]: Shouldn't this be `{{newVersion}}`?

File: release.js [hunk 2]
```diff
@@ -266,9 +266,9 @@
...
+           'sed -i "s,base href=\",base href=\"/HEAD,g" dist/docs/index.html'
```
Comment (topherfangio) [inline release.js]: Why was this changed from `{{newVersion}}` to `HEAD`?

**Comment by topherfangio** [comment]
@robertmesserle So, just to make sure I understand: the reason this wasn't working is that it was not properly escaped?

Also, just noting that this references (and should fix) #6695.

File: release.js (hunk 1 above)
Comment (robertmesserle) [inline release.js]: I believe so, yes.

File: release.js (hunk 2 above)
Comment (robertmesserle) [inline release.js]: Should be {{newVersion}} as well

**Comment by robertmesserle** [comment]
@topherfangio Yea, looks like it's just an escaping bug.  I'm not sure why it STOPPED working (since this used to work), so I just pulled the lines that are working from the snapshop script and properly escaped them.

**Comment by topherfangio** [comment]
@robertmesserle Awesome. Are you able to test this somehow without pushing a formal release?

**Comment by robertmesserle** [comment]
Not right now, but I can tomorrow.

**Comment by robertmesserle** [comment]
@jelbourn This should be ready, please confirm by performing a dry-run on your machine.

File: release.js [hunk 3]
```diff
@@ -331,6 +331,25 @@
     }
   }
 
+  function replaceFilePaths () {
```
Comment (jelbourn) [inline release.js]: This needs some comments explaining what's going on and why you're doing it.

File: release.js [hunk 4]
```diff
@@ -331,6 +331,25 @@
...
+    var path = __dirname + '/dist/docs/docs.js';
+    var file = fs.readFileSync(path);
+    var contents = file.toString()
+        .replace(/http:\/\/localhost:8080\/angular-material/g, 'https://cdn.gitcdn.xyz/cdn/angular/bower-material/v{{newVersion}}/angular-material')
+        .replace(/http:\/\/localhost:8080\/docs.css/g, 'https://material.angularjs.org/{{newVersion}}/docs.css');
+    fs.writeFileSync(path, contents);
+
+  }
+
+  function replaceBaseHref (folder) {
```
Comment (jelbourn) [inline release.js]: Comments here also.

**Comment by jelbourn** [comment]
@robertmesserle where should I look to make sure the issue is fixed?

**Comment by robertmesserle** [comment]
@jelbourn You'll want to make sure that:
- `base href` is correct in `latest/index.html` and `1.0.x/index.html`
- `localhost` is not found anywhere in `latest/docs.js` as well as `1.0.x/docs.js`
- run `git status` from the `code.material.angularjs.org` folder to make sure there are no uncommitted changes

That **should** be it.
